router.go

```diff
@@ -125,10 +135,6 @@ func (app *App) handler(rctx *fasthttp.RequestCtx) {
 	if match && app.Settings.ETag {
 		setETag(ctx, false)
 	}
-	// Scan stack for other methods
-	if !match {
```

Comment:
```go
if ctx.Fasthttp.Response.StatusCode() == StatusNotFound {
        setMethodNotAllowed(ctx)
    }
```
that should suffice

Reply:
While a file is not found, the status code also can be 404.

Reply:
Already removed the`ctx.matched` things.